delta-app/src/main/java/io/cdap/delta/app/DeltaContext.java

```diff
@@ -195,7 +204,7 @@ public void notifyFailed(Throwable cause) {
   }
 
   void clearMetrics() {
-    eventMetrics.clear();
+    tableEventMetrics.clear();
```

Comment:
this should loop through the EventMetrics and clear each individual one.

Reply:
Any reason for looping through it?

Reply:
actually, not needed if EventMetrics.clear() doesn't do anything except clear in-memory things.